mesonbuild/mdist.py

```diff
@@ -56,11 +56,12 @@ def create_hash(fname):
 
 
 def del_gitfiles(dirname):
-    for f in glob(os.path.join(dirname, '.git*')):
-        if os.path.isdir(f) and not os.path.islink(f):
-            windows_proof_rmtree(f)
-        else:
-            os.unlink(f)
+    for f in [os.path.join(dirname, '.git'), os.path.join(dirname, '.gitattributes'), os.path.join(dirname, '.gitignore'), os.path.join(dirname, '.gitmodules')]:
```

Comment:
Rather than joining every time you should use `os.path.split()` to get the last component and then comparing that against the options.

Reply:
You mean like:

```python
for f in glob(os.path.join(dirname, 'git*'))
  if os.path.split(f).tail in ['.git', '.gitattribute', '.gitignore', '.gitmodules']
    ...
```

(sorry I'm on my phone, not sure if `os.path.split().tail` is actually doable)

Reply:
Basically `os.path.split(f)[1]`. You should build the array before the loop and store it in a variable, otherwise Python will recreate the object on each iteration. Since it is immutable you should use a tuple (`('foo', 'bar')`) rather than an array (`['foo', 'bar']`).